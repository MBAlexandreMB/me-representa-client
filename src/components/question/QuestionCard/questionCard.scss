@import url('https://fonts.googleapis.com/css2?family=Lato&display=swap');

.questioncard-container {
  font-family: 'Lato';
  position: relative;
  width: 500px;
  padding: 20px 20px 40px;
  box-shadow: 1px 1px 5px 1px #ccc;
  border-radius: 5px;

  h3 {
    margin: 0;
  }

  p {
    margin-bottom: 3px;
  }

  a {
    text-decoration: none;
    color: rgba(0, 0, 0, .5);
    font-size: 0.875rem;
  }

  .buttons-container {
    display: flex;
    justify-content: center;

    button {
      margin: 20px 10px 0;
    }
  }

  .skip-btn {
    position: absolute;
    bottom: 20px;
    right: 20px;

    cursor: pointer;
    font-size: 0.875rem;
    color:rgba(0, 0, 0, .5);
  }
}