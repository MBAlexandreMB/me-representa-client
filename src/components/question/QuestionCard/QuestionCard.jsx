import React from "react";
import './questionCard.scss'; 

const QuestionCard = ({ data, onSkip, onOptionChosen }) => {
  return (
    <>
    {
      data ?
      <div className="questioncard-container">
        <p>{data.question}</p>
        <a className="proposition-link" href={data.link}>
          Link para a proposta
        </a>
        <div className="buttons-container">
          <button onClick={() => onOptionChosen('Sim')}>Sim</button>
          <button onClick={() => onOptionChosen('Não')}>Não</button>
          <button onClick={() => onOptionChosen('Abstenção')}>Abstenção</button>
          <button onClick={() => onOptionChosen('Obstrução')}>Obstrução</button>
        </div>
        <p onClick={onSkip} className="skip-btn">Pular</p>
      </div>
      :
      <div className="questioncard-container">
        <h3>Nada para ser respondido!</h3>
      </div>
    }
    </>
  );
};

export default QuestionCard;
