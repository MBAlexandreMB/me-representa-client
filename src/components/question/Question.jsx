import React, { useState, useEffect } from 'react';
import QuestionCard from './QuestionCard/QuestionCard';
import votingService from '../../services/voting.service';

const Question = ({ ...restProps }) => {
  const [data, setData] = useState();
  const [answered, setAnswered] = useState({});
  
  useEffect(() => {
    handleCardChange(Object.keys(answered));
  }, [answered]);

  const handleCardChange = (answeredQuestions) => {
    votingService.getRandomQuestion(answeredQuestions)
    .then((question) => {
      if (question) {
        setData({
          linkedVoting: question.votacaoVinculada,
          question: question.descricao, 
          link: ''
        })
      } else {
        setData(undefined);
      }
    });
  };

  const handleAnswer = (answer) => {
    const obj = { 
      ...answered,
      [data.linkedVoting]: answer,
    };

    votingService.getMatches(obj)
    .then(congressmen => {
      console.log('----------------');
      congressmen.forEach(element => {
        console.log(element.match);
      });
      console.log('Total:', congressmen.length);
      console.log('----------------');
    });

    setAnswered(obj);
  };

  return (
    <QuestionCard
      onSkip={handleCardChange}
      onOptionChosen={handleAnswer}
      data={data}
      {...restProps}
    />
  );
};

export default Question;