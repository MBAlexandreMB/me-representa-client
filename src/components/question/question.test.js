import React from 'react';
// import { render, unmountComponentAtNode } from "react-dom";
// import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';

import Question from './Question';
import QuestionCard from './QuestionCard/QuestionCard';

let container = null;
beforeEach(() => {
  // // setup a DOM element as a render target
  // container = document.createElement("div");
  // document.body.appendChild(container);
});

afterEach(() => {
  // // cleanup on exiting
  // unmountComponentAtNode(container);
  // container.remove();
  // container = null;
});

describe("Question Container Component", () => {
  it('should exist', () => {
    expect(typeof Question).toBe('function');
  });

  // it('should render a card', () => {
  //   const component = renderer.create(<Question />).toJSON();
  //   const card = renderer.create(<QuestionCard />);
  //   console.log(card.toJSON());
  //   expect(component).toMatchSnapshot();
  // });
});