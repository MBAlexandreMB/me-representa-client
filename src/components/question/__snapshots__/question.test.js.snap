// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Container Component should render a card 1`] = `
<div
  className="questioncard-container"
>
  <h3>
    Nada para ser respondido!
  </h3>
</div>
`;
